pkgs/applications/editors/emacs/make-emacs.nix

```diff
@@ -185,20 +185,20 @@ mkDerivation (finalAttrs: {
     # Reduce closure size by cleaning the environment of the emacs dumper
     ''
       substituteInPlace src/Makefile.in \
-        --replace 'RUN_TEMACS = ./temacs' 'RUN_TEMACS = env -i ./temacs'
+        --replace-fail 'RUN_TEMACS = ./temacs' 'RUN_TEMACS = env -i ./temacs'
     ''
 
     ''
       substituteInPlace lisp/international/mule-cmds.el \
-        --replace /usr/share/locale ${gettext}/share/locale
-
+        --replace-fail /usr/share/locale ${gettext}/share/locale
+    ''
+  ] ++ lib.optionals (lib.versionOlder version "29") [
+    ''
       for makefile_in in $(find . -name Makefile.in -print); do
-        substituteInPlace $makefile_in --replace /bin/pwd pwd
+        substituteInPlace $makefile_in --replace-fail /bin/pwd pwd
       done
     ''
-
-    ""
```

Comment:
nit(non-blocking): removing this empty string is not part of `use --replace-fail`.